meta/autoid/autoid.go

```diff
@@ -137,13 +137,12 @@ func (alloc *allocator) Rebase(tableID, requiredBase int64, allocIDs bool) error
 		if allocIDs {
 			if alloc.isUnsigned {
```

Comment:
You want to prevent overflow? 
Should we 
```
if uNewBase > (math.Uint64 - step) {
   panic
}
```

Reply:
Yes,
but why we panic here? @tiancaiamao

Reply:
Well, maybe error or warning...
I mean, if the ID overflow, it should not be changed to math.Uint64

Reply:
If alloc.end overflow, what will happen?

Reply:
@tiancaiamao 
The result of TiDB master:
``` sql
tidb> create table t(a bigint signed not null auto_increment, unique key(a));
Query OK, 0 rows affected (0.02 sec)

tidb> insert into t value(9223372036854775806);
Query OK, 1 row affected (0.00 sec)

tidb> insert into t value();
Query OK, 1 row affected (0.00 sec)

tidb> insert into t value();
Query OK, 1 row affected (0.00 sec)

tidb> insert into t value();
Query OK, 1 row affected (0.00 sec)

tidb> insert into t value();
Query OK, 1 row affected (0.00 sec)

tidb> select * from t;
+----------------------+
| a                    |
+----------------------+
| -9223372036854775808 |
| -9223372036854775806 |
| -9223372036854775804 |
| -9223372036854775802 |
|  9223372036854775806 |
+----------------------+
5 rows in set (0.00 sec)
```

Reply:
Overwrite the old row id is dangerous for data consistency.
Is it acceptable to change the behavior to throw error?

Reply:
As a tradeoff, https://github.com/pingcap/tidb/pull/8181#pullrequestreview-174023063 may be better?

Reply:
I think throw error is simple, easier, and more robust.